server/services/roService.ts

```diff
@@ -22,8 +24,25 @@ export class RoService {
     }
   }
 
-  private filterMostRecentActiveBooking(offenderSentences: OffenderSentence[]): OffenderSentence[] {
-    return offenderSentences.filter((o) => o.mostRecentActiveBooking === true)
+  getLatestSentences(offenderSentences: OffenderSentence[]): OffenderSentence[] {
+    const groupedOffenderSentencess = groupBy(offenderSentences, ({ offenderNo }) => offenderNo)
+    let latestOffenderSentences = []
+
+    Array.from(groupedOffenderSentencess.values()).flatMap((sentences) => {
+      const hasNoDatesToCompare = sentences.find((b) => absentDatesToCompare(b))
+      if (hasNoDatesToCompare) {
+        sentences.forEach((sentence) => latestOffenderSentences.push(sentence))
+      } else {
+        const sortedSentences = sentences.sort((a, b) => {
+          return moment(
+            a.sentenceDetail.topupSupervisionExpiryCalculatedDate || a.sentenceDetail.licenceExpiryCalculatedDate
+          ).diff(b.sentenceDetail.topupSupervisionExpiryCalculatedDate || b.sentenceDetail.licenceExpiryCalculatedDate)
+        })
+        const latestSentence = sortedSentences.pop()
+        latestOffenderSentences.push(latestSentence)
+      }
+    })
+    return latestOffenderSentences
```

Comment:
New criteria for showing the latest sentences is:
- show the sentence with the latest calculated TUSED if present
- if not show the sentence with the latest calculated LED
- if neither of these dates are present then we don't think we are able to identify the latest sentence so show all the sentences for that offender (this should be a rare edge case and showing all sentences is current behaviour anyway).